lib/logger/lib/logger/watcher.ex

```diff
@@ -28,18 +26,24 @@ defmodule Logger.Watcher do
   Removes the given handler.
   """
   def unwatch(mod, handler) do
-    GenEvent.remove_handler(mod, handler, :ok)
+    result = :gen_event.delete_handler(mod, handler, :ok)
+    _ = Supervisor.terminate_child(@name, {__MODULE__, {mod, handler}})
+    result
```

Comment:
There is this scenario:
* handler crashes
* attempt to delete handler returns `{:error, :module_not_found}`
* supervisor restarts handler
* call to terminate stops watcher
* returns error

Since the supervisor is "our source of truth" should we just call `Supervisor.terminate_child/2` since it provides the same guarantee?